### PR TITLE
App development post part II

### PR DESCRIPTION
**Pull request by cgebken**

### CODE REVIEW

File: _posts/2019/2019-12-16-how-to-enhance-an-application-using-firebase-auth-and-firebase-functions.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+Lets start with initializing these Cloud Functions.
+In our repository, we again run `firebase init`, but this time we select `functions`.
+For this example, we now select JavaScript, but Typescript would also be a valid option.
+As a result, all default npm dependencies get installed.
+Now, we have a new `functions` directory in our repository.
+At this point, we could dive deeper into the theory of Firebase functions, but we'll skip that part for now.
+
+Instead, we use the Firebase [HTTPS callables](https://firebase.google.com/docs/functions/callable){:target="_blank"} and trigger them with the Javascript Firebase SDK.
+If you are interested in the methods we use for that, check our [Javascript Tutorial](http://docs.beyondshop.cloud/#_tutorial_javascript){:target="_blank"}.
```
Comment (cgebken) [inline _posts/2019/2019-12-16-how-to-enhance-an-application-using-firebase-auth-and-firebase-functions.md]: Todo: Link needs to be adapted once anchors are working in new documentation.

File: _posts/2019/2019-12-16-how-to-enhance-an-application-using-firebase-auth-and-firebase-functions.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+  ...,
+  "engine": {
+    "node": "8"
+  }
+}
+```
+
+Now, all functions will be deployed as Node 8 functions.
+
+Our function will also use the `getBeyondAuthToken` method introduced in the [Javascript tutorial](http://docs.beyondshop.cloud/#_tutorial_javascript){:target="_blank"} to fetch the API tokens.
```
Comment (cgebken) [inline _posts/2019/2019-12-16-how-to-enhance-an-application-using-firebase-auth-and-firebase-functions.md]: Todo: Link needs to be adapted once anchors are working in new documentation.

File: _posts/2019/2019-12-16-how-to-enhance-an-application-using-firebase-auth-and-firebase-functions.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+The document does not need any properties yet.
+
+We'll now use the newly created collection `users` to save our tokens after the user logged in.
+First, we have to add the Firestore API to the `public/index.html`:
+
+```html
+  <script defer src="/__/firebase/7.2.0/firebase-firestore.js"></script>
+```
+
+Next, we can use it from our existing script after we `fetch` the access data via the API.
```
Comment (cgebken) [inline _posts/2019/2019-12-16-how-to-enhance-an-application-using-firebase-auth-and-firebase-functions.md]: I didn't really get what you mean with "it". Could you give me some insights Florian ?:)

Comment (Raigen) [reply _posts/2019/2019-12-16-how-to-enhance-an-application-using-firebase-auth-and-firebase-functions.md]: First, we have to add **the Firestore API** to the `public/index.html`:
Next, we can use **it** from our existing script after we `fetch` the access data via the API.

**Comment by cgebken** [comment]
@Raigen Ready for final review :)

File: _posts/2019/2019-12-16-how-to-enhance-an-application-using-firebase-auth-and-firebase-functions.md [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+In this part, we'll take care of a database creation and security improvements.
+
+## Step 1 - Database creation
+
+Via the authentication service Firebase Auth, we receice tokens for different shops.
+Only the specific merchant of the shop should have access to this token.
+To do so, we first need to create a database in the Firebase console (Develop > Database).
+Then, we create a Cloud Firestore database.
+For fast results, we choose the test mode for now.
+We'll secure the data in a later step.
```
Comment (Raigen) [inline _posts/2019/2019-12-16-how-to-enhance-an-application-using-firebase-auth-and-firebase-functions.md]: I just recognized that I never wrote about this step :/

File: _posts/2019/2019-12-16-how-to-enhance-an-application-using-firebase-auth-and-firebase-functions.md [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+
+In the [last part of this series](/blog/coding/how-to-bootstrap-an-application-with-authentication-using-firebase/){:target="_blank"}, we went through all the steps required to bootstrap an application with authentication.
+Our intermediate result was a login UI that can be used to distinguish users.
+Furthermore, we were able to give API access only to registered users.
+In this part, we'll take care of a database creation and security improvements.
+
+## Step 1 - Database creation
+
+Via the authentication service Firebase Auth, we receice tokens for different shops.
+Only the specific merchant of the shop should have access to this token.
```
[the remark this answers is not in the document]
Comment (cgebken) [reply _posts/2019/2019-12-16-how-to-enhance-an-application-using-firebase-auth-and-firebase-functions.md]: Ahhh... I thought I got rid of merchants. Good catch. Adapted ;)

File: _posts/2019/2019-12-16-how-to-enhance-an-application-using-firebase-auth-and-firebase-functions.md (hunk 4 above)
Comment (cgebken) [reply _posts/2019/2019-12-16-how-to-enhance-an-application-using-firebase-auth-and-firebase-functions.md]: Mmh... Made a proposal how to handle this... Or would you vote for not mentioning it at all? :D

**Pull request merged**